app/src/main/java/com/app/spott/activities/MainActivity.java

```diff
@@ -32,25 +34,28 @@ protected void onCreate(Bundle savedInstanceState) {
         }
 
     }
+
     // Get the userId from the cached currentUser object
     void startWithCurrentUser() {
         ParseUser loggedInUser = ParseUser.getCurrentUser();
-        User user;
-        try {
-            user = User.getByOwner(loggedInUser);
-        } catch (ParseException e) {
-            e.printStackTrace();
-            user = setupNewUser();
-        }
+        User.getByOwner(loggedInUser, new GetCallback<User>() {
+            @Override
+            public void done(User object, ParseException e) {
+                if (object == null)
+                    currentUser = setupNewUser();
+                else
+                    currentUser = object;
+            }
+        });
 
-        Log.d("User", user.getFirstName());
+        Log.d("User", currentUser.getFirstName());
     }
 
-    private User setupNewUser(){
+    private User setupNewUser() {
         return null;
     }
 
-    private User setupAdil(){
+    private User setupAdil() {
```

Comment:
@sgzsh269 @aparnarsjain use something like this and it will linkup your user login with user model.. this is a one time and then we can get rid of this method